Documentation/BuildInstructions.md

```diff
@@ -30,6 +30,8 @@ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 900 --slave /
 #### macOS prerequisites
 Make sure you have all the dependencies installed:
 ```bash
+xcode-select --install
+xcode-select -s /Applications/Xcode.app/Contents/Developer #Requires sudo
```

Comment:
This should absolutely not be in build instructions

Reply:
It was a required prerequisite, that took some googling to figure out why things weren't working.

Reply:
How do you even install/use Homebrew without working command-line developer tools?

Reply:
I think it is fair to assume that our MacOS developers will come with Xcode, and home-brew installed on their computer (with nothing else).


I had Xcode, and home-brew installed, but I needed to download the Xcode command line tools before some of the brew packages would properly install (I think flock gave me issues).  I don't think the Xcode developer tools come standard with the Xcode installer from the AppStore, so that 1 line may save others a lot of time.


In order to get farther in the Toolchain build, I needed to run the `xcode-select -s <dir>` command to get the serenity build tool to recognize that I had the command-line developer tools installed (I was failing VERY early in the toolchain build).


I think we should add this to the instructions simply just to save others time, and make it easier to use (maybe in the notes section)?

Reply:
> I think we should add this to the instructions simply just to save others time, and make it easier to use (maybe in the notes section)?

We're not interested in saving others time. It might be shocking, but we want people who want to contribute to Serenity to at least know how to build C++ code on their systems.

Reply:
Thanks, I am going to find another project to contribute too.

Reply:
@seze4251 if you've already moved on that's understandable, but I can assure you not everyone involved in this project shares that opinion - it should very much be possible for people not coming from a C++ background to build, run and try SerenityOS on macOS.

Please consider re-opening this PR, @bugaevc does not have the final saying about this topic.

Reply:
Hey now @bugaevc let's chillax a little bit here. Mac users are people too, I've been one myself for over a decade! :)

It's fine to spell things out in the build instructions. What we don't want is one-click installs / ISO images / etc, things that invite non-technical people to create unwanted noise.

Reply:
Whoa... 😟

Hey, it's not my opinion; it's [your own](https://github.com/SerenityOS/serenity/pull/2313#issuecomment-631966935), @awesomekling, at least the way I understood it. I, for one, would welcome making the instructions friendlier (if they're not friendly enough yet), but as Linus says, I don't make decisions about that.

If course Mac users are people; that's why we have macOS build instructions. But hardcoding a path to Xcode in build instructions, or including commands that change system settings in build instructions, is not the way to go. Consider this: people may (and should) have already have their system set up properly, with `xcode-select` pointing to another path.

And again, we would expect and want people to at least know how to build C++ code on their machines prior to contributing to the system (*that* is my opinion that I hope you share). And a properly set.up toolchain is a requirement for using Homebrew. I don't think we should provide instructions for setting up a Darwin system for development from scratch, should we?

What we could do, I guess, is add a note like "hey" if your toolchain is not set up, you might need to run `xcode-select --install` or `xcode-select --switch <path>`".

Reply:
> Hey, it's not my opinion; it's [your own](https://github.com/SerenityOS/serenity/pull/2313#issuecomment-631966935), @awesomekling, at least the way I understood it. I, for one, would welcome making the instructions friendlier (if they're not friendly enough yet), but as Linus says, I don't make decisions about that.

You've misunderstood my opinion. I'm specifically against turnkey solutions like Docker because they encourage superficial engagement and "drive-by development."

I'm all for making the instructions as friendly as possible, as long as we keep the proof-of-work mechanism where people have to make a tiny personal effort to get up and running. :)

> But hardcoding a path to Xcode in build instructions, or including commands that change system settings in build instructions, is not the way to go. Consider this: people may (and should) have already have their system set up properly, with `xcode-select` pointing to another path.

Yes, that's a technical issue that can be expanded upon in a macOS section of the build instructions, or perhaps even a dedicated document about building/developing on macOS.

> And again, we would expect and want people to at least know how to build C++ code on their machines prior to contributing to the system (_that_ is my opinion that I hope you share). And a properly set up toolchain is a requirement for using Homebrew. I don't think we should provide instructions for setting up a Darwin system for development from scratch, should we?

Obviously this project is not appropriate for beginners and our documentation should be written assuming the reader has multiple years of active programming experience.

> What we could do, I guess, is add a note like "hey, if your toolchain is not set up, you might need to run `xcode-select --install` or `xcode-select --switch <path>`".

Sure, that sounds very reasonable.

Reply:
@seze4251, I'm sorry if I sounded harsh, I didn't mean to. And I certainly didn't want to discourage you from contributing to the project. Apparently I misrepresented our official stance on the "entrance time fee"; sorry about that, too.

> I had Xcode, and home-brew installed, but I needed to download the Xcode command line tools before some of the brew packages would properly install (I think flock gave me issues).

Exactly; you need command-line tools before you can properly use Homebrew. And Homebrew's install script [already includes logic for installing them](https://github.com/Homebrew/install/blob/master/install.sh#L461-L490).

It seems nobody disagrees with adding a tip about `xcode-select` in a separate section / notes, to help people workaround weird cases where they have Homebrew but not a working toolchain.

But I maintain it should not be a part of the larger block of commands people are more-or-less expected to just copy-paste, and it should not tell people to use `/Applications/Xcode.app/Contents/Developer` specifically — it's fine to suggest that as one path that's likely to work, but not as the one definitive path to use.